### PR TITLE
Consolidate temporal unit pickers between dashboards and timeseries chrome

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/42118
Demo https://www.loom.com/share/b248f95894da47adbc7558e0122ede35

Extracts a base component from `TimeseriesChrome` and uses it in `TemporalUnitWidget` for parameters.

Screenshot 2024-06-12 at 10 32 12

### CODE REVIEW

File: frontend/src/metabase/querying/components/TimeseriesChrome/TimeseriesBucketPicker/TimeseriesBucketPicker.tsx [hunk 1]
```diff
@@ -14,97 +14,97 @@ interface TimeseriesBucketPickerProps {
...
+    </Popover>
   );
 }
 
-function isInitiallyExpanded(items: BucketItem[], selectedItem: BucketItem) {
-  const canExpand = items.length > INITIALLY_VISIBLE_ITEMS_COUNT;
-  if (!canExpand || !selectedItem) {
-    return false;
-  }
+interface TimeseriesBucketDropdownProps {
```
Comment (ranquild) [inline frontend/src/metabase/querying/components/TimeseriesChrome/TimeseriesBucketPicker/TimeseriesBucketPicker.tsx:59]: It's always a good practice to any calls (especially MBQL lib) when the dropdown is actually opened.

File: frontend/src/metabase/parameters/components/widgets/TemporalUnitWidget/TemporalUnitWidget.tsx [hunk 2]
```diff
@@ -19,22 +17,21 @@ export function TemporalUnitWidget({
 }: TemporalUnitWidgetProps) {
   const availableUnits =
     parameter.temporal_units ?? Lib.availableTemporalUnits();
+  const availableItems = availableUnits.map(unit => ({
```
Comment (uladzimirdev) [inline frontend/src/metabase/parameters/components/widgets/TemporalUnitWidget/TemporalUnitWidget.tsx:20]: memo?

File: frontend/src/metabase/querying/components/TemporalUnitPicker/TemporalUnitPicker.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+interface TemporalUnitPickerProps {
+  value: TemporalUnit | undefined;
+  availableItems: TemporalUnitItem[];
+  canRemove?: boolean;
+  onChange: (newValue: TemporalUnit) => void;
+  onRemove?: () => void;
+}
+
+export function TemporalUnitPicker({
+  value,
```
Comment (uladzimirdev) [inline frontend/src/metabase/querying/components/TemporalUnitPicker/TemporalUnitPicker.tsx:24]: should we have a test when `value` is not defined? is it possible when no value is selected only?

**Review by uladzimirdev — APPROVED** [review]

File: frontend/src/metabase/querying/components/TemporalUnitPicker/TemporalUnitPicker.tsx (hunk 3 above)
Comment (ranquild) [reply frontend/src/metabase/querying/components/TemporalUnitPicker/TemporalUnitPicker.tsx:24]: Added tests

File: frontend/src/metabase/parameters/components/widgets/TemporalUnitWidget/TemporalUnitWidget.tsx (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase/parameters/components/widgets/TemporalUnitWidget/TemporalUnitWidget.tsx:20]: Not needed - this is a short lived component in a popover

**Pull request merged**